github/activity_notifications.go

```diff
@@ -97,15 +97,15 @@ func (s *ActivityService) ListRepositoryNotifications(ctx context.Context, owner
 }
 
 type markReadOptions struct {
-	LastReadAt Timestamp `json:"last_read_at,omitempty"`
+	LastReadAt *Timestamp `json:"last_read_at,omitempty"`
```

Comment:
@gmlewis  I found this case where the documentation says that the value can be omitted. I think this should be a pointer (as well as other cases). And also the "read" field is missing. (Github docs: https://docs.github.com/en/rest/activity#mark-notifications-as-read)

In going through the tests I found some interesting cases. Do you think it is better if I open an issue for each one, to talk more about each one?